problem_builder/mentoring.py

```diff
@@ -776,3 +793,98 @@ def workbench_scenarios():
+        display_name=_("Title (Display name)"),
+        help=_("Title to display"),
+        default=_("Mentoring Questions (with explicit steps)"),
+        scope=Scope.settings
+    )
+
+    editable_fields = ('display_name',)
+
+    @lazy
+    def questions(self):
+        """ Get the usage_ids of all of this XBlock's children that are "Questions" """
+        return list(chain.from_iterable(self.runtime.get_block(step_id).steps for step_id in self.steps))
+
+    @property
+    def steps(self):
+        """
+        Get the usage_ids of all of this XBlock's children that are "Steps"
+        """
+        from .step import MentoringStepBlock  # Import here to avoid circular dependency
+        return [
+            _normalize_id(child_id) for child_id in self.children if
+            child_isinstance(self, child_id, MentoringStepBlock)
+        ]
+
+    def student_view(self, context):
+        fragment = Fragment()
+        child_content = u""
+
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if child is None:  # child should not be None but it can happen due to bugs or permission issues
+                child_content += u"<p>[{}]</p>".format(self._(u"Error: Unable to load child component."))
+            elif not isinstance(child, MentoringMessageBlock):
+                child_fragment = self._render_child_fragment(child, context, view='mentoring_view')
+                fragment.add_frag_resources(child_fragment)
+                child_content += child_fragment.content
+
+        fragment.add_content(loader.render_template('templates/html/mentoring.html', {
+            'self': self,
+            'title': self.display_name,
+            'show_title': self.show_title,
+            'child_content': child_content,
+        }))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder.css'))
+
+        self.include_theme_files(fragment)
+
+        return fragment
+
+    @property
+    def allowed_nested_blocks(self):
+        """
+        Returns a list of allowed nested XBlocks. Each item can be either
+        * An XBlock class
+        * A NestedXBlockSpec
+
+        If XBlock class is used it is assumed that this XBlock is enabled and allows multiple instances.
+        NestedXBlockSpec allows explicitly setting disabled/enabled state, disabled reason (if any) and single/multiple
+        instances
+        """
+        from .step import MentoringStepBlock  # Import here to avoid circular dependency
+        return [
+            MentoringStepBlock,
+            NestedXBlockSpec(CompletedMentoringMessageBlock, boilerplate='completed'),
+            NestedXBlockSpec(IncompleteMentoringMessageBlock, boilerplate='incomplete'),
+            NestedXBlockSpec(MaxAttemptsReachedMentoringMessageBlock, boilerplate='max_attempts_reached'),
+            NestedXBlockSpec(OnAssessmentReviewMentoringMessageBlock, boilerplate='on-assessment-review'),
```

Comment:
@itsjeyd if I do recall right, we decided that messages should be at step level. Also, if the reason you have created a dedicated class for each message is to support names, it could be done "better" in two ways:

1. Create shims instead of actual subclasses
2. Even better: add ability to override block name in NestedXblockSpec in xblock-utils. It is edx repo, but (at least so far), the master is considered an active-development branch, so we don't go into upstream review with changes - we only send batches of changes to edx-release (which is what edx uses) branch from time to time.